### PR TITLE
CaffeLoader supports multiple inputs and some additional outputs

### PR DESCRIPTION
**Pull request by SeaOfOcean**
## What changes were proposed in this pull request?

CaffeLoader supports multiple inputs and some additional outputs

## How was this patch tested?

manual test with faster rcnn model

## Related links or issues (optional)
fixed https://github.com/intel-analytics/BigDL/issues/1691

### CODE REVIEW

File: spark/dl/src/main/scala/com/intel/analytics/bigdl/utils/caffe/CaffeLoader.scala [hunk 1]
```diff
@@ -262,19 +262,67 @@ class CaffeLoader[T: ClassTag](prototxtPath: String, modelPath: String,
...
+    addInputList(inputs, layers, top2LayerMap, layersMap)
+  }
+
+  private def addInputList(inputs: Seq[ModuleNode[T]],
+    layers: ArrayBuffer[ModuleNode[T]],
+    top2LayerMap: mutable.HashMap[String, String],
+    layersMap: mutable.HashMap[String, ModuleNode[T]]): Boolean = {
+    if (null != inputs) {
+      inputs.foreach(input => {
+        top2LayerMap(input.element.getName()) = input.element.getName()
```
Comment (wzhongyuan) [inline spark/dl/src/main/scala/com/intel/analytics/bigdl/utils/caffe/CaffeLoader.scala]: So 'top' for input layer will be always the same as its name ?

Comment (wzhongyuan) [reply spark/dl/src/main/scala/com/intel/analytics/bigdl/utils/caffe/CaffeLoader.scala]: Got it from you conveter logic below :)

**Comment by wzhongyuan** [comment]
LGTM

**Comment by SeaOfOcean** [comment]
pr validate pass

**Pull request merged**